src/frogger.cpp

```diff
@@ -21,32 +21,72 @@ int get_simulation_years(const Rcpp::List demp, SEXP r_sim_years) {
+    age_groups_hiv = 9;
+    data = projection_parameters["hAG_SPAN_coarse"];
+  } else {
+    Rcpp::stop(
+        "Invalid HIV age stratification must be 'full' or 'coarse' got '%s'.",
+        hiv_age_stratification);
+  }
+  if (LENGTH(data) != age_groups_hiv) {
+    Rcpp::stop("Invalid size of data, expected %d got %d", age_groups_hiv,
+               LENGTH(data));
+  }
+  TensorMap1<int> age_groups_hiv_span(INTEGER(data), age_groups_hiv);
+  return age_groups_hiv_span;
+}
+
 // [[Rcpp::export]]
-Rcpp::List run_base_model(const Rcpp::List demp, SEXP sim_years) {
-  const int proj_years = get_simulation_years(demp, sim_years);
+Rcpp::List run_base_model(const Rcpp::List data,
+                          const Rcpp::List projection_parameters,
+                          SEXP sim_years,
+                          std::string hiv_age_stratification = "full") {
+  const int proj_years = get_simulation_years(data, sim_years);
   const int num_genders = 2;
   const int age_groups_pop = 81;
   const int fertility_first_age_group = 15;
   const int age_groups_fert = 35;
+  const int disease_stages = 7;
+  const int treatment_stages = 3;
+  const int hiv_adult_first_age_group = 15;
+  const int time_art_start = *INTEGER(projection_parameters["t_ART_start"]) -
+                             1;  // 0-based indexing vs R 1-based
+  auto age_groups_hiv_span =
+      get_age_groups_hiv_span(projection_parameters, hiv_age_stratification);
+  int age_groups_hiv = age_groups_hiv_span.size();
 
-  TensorMap2<double> base_pop(REAL(demp["basepop"]), age_groups_pop,
+  TensorMap2<double> base_pop(REAL(data["basepop"]), age_groups_pop,
                               num_genders);
   // Survival has size age_groups_pop + 1 as this is the probability of
   // surviving between ages, so from 0 to 1, 1 to 2, ..., 79 to 80+ and
   // 80+ to 80+
-  TensorMap3<double> survival(REAL(demp["Sx"]), age_groups_pop + 1, num_genders,
+  TensorMap3<double> survival(REAL(data["Sx"]), age_groups_pop + 1, num_genders,
                               proj_years);
-  TensorMap3<double> net_migration(REAL(demp["netmigr_adj"]), age_groups_pop,
+  TensorMap3<double> net_migration(REAL(data["netmigr_adj"]), age_groups_pop,
                                    num_genders, proj_years);
-  TensorMap2<double> age_sex_fertility_ratio(REAL(demp["asfr"]),
+  TensorMap2<double> age_sex_fertility_ratio(REAL(data["asfr"]),
                                              age_groups_fert, proj_years);
-  TensorMap2<double> births_sex_prop(REAL(demp["births_sex_prop"]), num_genders,
+  TensorMap2<double> births_sex_prop(REAL(data["births_sex_prop"]), num_genders,
                                      proj_years);
 
   Parameters<double> params = {num_genders,
                                age_groups_pop,
                                fertility_first_age_group,
                                age_groups_fert,
+                               age_groups_hiv,
```

Comment:
This was the line that was triggering the `warning: narrowing conversion of 'age_groups_hiv' from 'long int' to 'int' [-Wnarrowing]` warning. I fixed by changing line 64 from 

```
auto age_groups_hiv = age_groups_hiv_span.size();
```
to
```
int age_groups_hiv = age_groups_hiv_span.size();
```
which makes the cast explicit?